src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -2271,6 +2275,50 @@ bool AdaptDispatch::SetDefaultBackground(const DWORD dwColor)
     return SetColorTableEntry(TextColor::DEFAULT_BACKGROUND, dwColor);
 }
 
+// Method Description:
+// DECAC - Assigns the foreground and background color indexes that should be
+//   used for a given aspect of the user interface.
+// Arguments:
+// - item: The aspect of the interface that will have its colors altered.
+// - fgIndex: The color table index to be used for the foreground.
+// - bgIndex: The color table index to be used for the background.
+// Return Value:
+// True if handled successfully. False otherwise.
+bool AdaptDispatch::AssignColor(const DispatchTypes::ColorItem item, const VTInt fgIndex, const VTInt bgIndex)
+{
+    auto targetColors = std::pair<ColorAlias, ColorAlias>{};
+    switch (item)
+    {
+    case DispatchTypes::ColorItem::NormalText:
+        targetColors = { ColorAlias::DefaultForeground, ColorAlias::DefaultBackground };
+        break;
+    case DispatchTypes::ColorItem::WindowFrame:
+        targetColors = { ColorAlias::FrameForeground, ColorAlias::FrameBackground };
+        break;
+    default:
+        return false;
+    }
+
+    if (fgIndex < TextColor::TABLE_SIZE)
+    {
+        _renderSettings.SetColorAliasIndex(targetColors.first, fgIndex);
+    }
+    if (bgIndex < TextColor::TABLE_SIZE)
+    {
+        _renderSettings.SetColorAliasIndex(targetColors.second, bgIndex);
+    }
```

Comment:
Is it possible to have these two if conditions inside the switch and drop `targetColors`?

Why is fg/bgIndex limited to `TextColor::TABLE_SIZE` (266) and not to 16 or 256?

Reply:
> Is it possible to have these two if conditions inside the switch and drop `targetColors`?

I did try that, but I found the extra bloat just made the code harder to follow. If we were allowed single line conditions, it would be different, but these 4-line monsters are just too much to be duplicated unnecessarily.

I also considered adding something like a `SetCheckedColorAliasIndex` to the `RenderSettings`, and then we could easily put everything inside the switch. But I thought maybe I should wait to see if we need this pattern elsewhere before going that route. That said, I'm not at all opposed to doing that now, if others think that is preferable.

> Why is fg/bgIndex limited to `TextColor::TABLE_SIZE` (266) and not to 16 or 256?

Because the defaults are all in the 256+ range. On startup, your default text color mapping is going to be the equivalent of `DECAC 1;261;262` and your default tab color mapping would be `DECAC 2;263;264`. If you wanted your tab color to match the default background color, you could do `DECAC 2;261;262`, and then any changes to the default colors would automatically be reflected in the tab as well (the effect shown in #702).

Ideally I would prefer apps didn't need to know those exact numbers, and they could query the initial `DECAC` settings to determine them. But I haven't implemented the queries yet, because there are issues there I need to figure out first. That's one of the reasons this is still a draft, but I may just leave that for a followup PR.

Reply:
Now that I think about it, we might as well make `SetColorAliasIndex` always check the index, and then there'll be no problem getting rid of the `targetColors`. In most other places we're passing `SetColorAliasIndex` a known safe constant, but the compiler should be able to skip the test for those cases. I'll give that a try.

Reply:
I love that idea!